File: client/src/ts/packets/reposition.ts
import Packet, {PacketType} from "./packet";

export default class RepositionPacket extends Packet {
    public x: number;
    public y: number;
    public id: number;

    constructor(packetDict: any) {
        super(PacketType.Reposition);

        this.x = packetDict['x'];
        this.y = packetDict['y'];
        this.id = packetDict['id'];
    }

    public dictify(): any {
        throw 5;
    }
}

File: server/network/packets/reposition.py
from server.network.packets import PacketType
from server.network.packets.packet import Packet


class RepositionPacket(Packet):
    def __init__(self, entity):
        self.entity = entity

    def dictify(self):
        return dict(type=self.get_type(), id=self.entity.id, x=self.entity.x, y=self.entity.y)

    def get_type(self):
        return PacketType.Reposition
